src/firefly/js/charts/ui/HistogramOptions.jsx

```diff
@@ -60,6 +60,41 @@ const algorithmOptions = [  {label: 'Bayesian blocks', value: 'bayesianBlocks'},
 const binSizeOptions = [  {label: 'Number of bins:', value: 'numBins'},
     {label: 'Bin width:', value: 'binWidth'} ];
 
+
+var columnNameReducer= (colValStats, histogramParams) => {
+    if (!colValStats) {
+        return {};
+    }
+    return (inFields, action) => {
+
+        if (!inFields) {
+            return {};
+        }
+        let fieldKey = undefined;
+        if (action.type === VALUE_CHANGE) {
+            // when column name changes, update the min/max input
+            fieldKey = get(action.payload, 'fieldKey');
+            const colName = action.payload.value;
+            if (fieldKey === 'columnOrExpr') {
+
+                for (var i=0; i<colValStats.length; i++){
+                    if (colName=== colValStats[i].name){
+                        const dataMin = colValStats[i].min;
+                        const dataMax = colValStats[i].max;
+                        const binWidth = (dataMax - dataMin)/50.0;
```

Comment:
What if the value of numBins is not 50? Perhaps, it should be divided by the current value of 'numBins' field.

Another improvement would be adjust the value of 'binWidth' when all of 'numBins', 'minCutoff', 'maxCutoff' are populated and one of them changes.